### PR TITLE
update dependencies except react

### PR DESCRIPTION
**Pull request by ToppleTheNun**
fully functioning conversion to use react-router v6 instead of react-router v5.

also updates all other dependencies except for the react 17 -> react 18 changes,
which will come in a separate PR.

other slightly notable changes are included in the commit messages throughout
the PR but are summarized here for reference:
- update lingui but pin it to 3.14.0 due to this issue: https://togithub.com/lingui/js-lingui/issues/1300
- update react-redux to 8.x and also use @reduxjs/toolkit to match recommended patterns
- update typescript (and eslint support) to typescript 4.9

### CODE REVIEW

File: src/interface/actions/reportHistory.ts
Comment (emallson) [inline src/interface/actions/reportHistory.ts]:
```suggestion
export interface ReportHistoryEntry {
```

File: src/interface/actions/reportHistory.ts
Comment (emallson) [inline src/interface/actions/reportHistory.ts]:
```suggestion
export function appendReportHistory(report: ReportHistoryEntry) {
```

minor type name change for clarity---this type is not the history, but an *entry* in the history.

File: src/interface/reducers/reportHistory.ts
Comment (emallson) [inline src/interface/reducers/reportHistory.ts]: will also need type renames here. seems like a "let langserver handle it"

**Review by emallson — CHANGES_REQUESTED** [review]
some minor changes. im going to do some manual testing before we merge this for my own sanity

**Review by emallson — APPROVED** [review]
time to sit back and watch the fireworks

**Comment by ToppleTheNun** [comment]
I plan to merge this tomorrow morning assuming that no conflicts or major issue arise before then.

**Pull request merged**
